packages/lib/src/interactions/utils.ts

```diff
@@ -99,3 +106,52 @@ export function clampRectangleToVis(
   const shift = newCenter.clone().sub(center);
   return [startPoint.clone().add(shift), endPoint.clone().add(shift)];
 }
+
+export function getDistinctPoints(
+  startPoint: Vector2,
+  endPoint: Vector2,
+  xDomain: Domain,
+  yDomain: Domain
+): [Vector2, Vector2] {
+  const [xMin, xMax] = xDomain;
+
+  if (Math.abs(endPoint.x - startPoint.x) < Math.abs(xMax - xMin) / 200) {
+    return [new Vector2(xMin, startPoint.y), new Vector2(xMax, endPoint.y)];
+  }
+
+  const [yMin, yMax] = yDomain;
+  if (Math.abs(endPoint.y - startPoint.y) < Math.abs(yMax - yMin) / 200) {
+    return [new Vector2(startPoint.x, yMin), new Vector2(endPoint.x, yMax)];
+  }
+
+  return [startPoint, endPoint];
+}
+
+export function getZoomTransform(
+  keepRatio?: boolean,
+  fullDimIfEmpty?: boolean
+): ZoomTransformFn {
```

Comment:
Not sure I really get the point of returning a function here. If it's the number of parameters, you can combine them: `dataSelection`, `visiblesDomains`. I would also rename the final function `getZoomSelection` or `getZoomArea`.